src/sphinx_autodoc_typehints/__init__.py

```diff
@@ -150,7 +169,10 @@ def format_annotation(annotation: Any, config: Config) -> str:
         formatted_args = f"\\[{', '.join(repr(arg) for arg in args)}]"
 
     if args and not formatted_args:
-        fmt = [format_annotation(arg, config) for arg in args]
+        try:
+            fmt = [format_annotation(arg, config) for arg in args]
+        except TypeError:
```

Comment:
Also not entirely sure if this exception handling needs to be refined.

Reply:
Why do we need this?

Reply:
To handle things with bare integers in annotations, like `Float[32]` in `nptyping` and (presumably) [`Literal[640]` in PEP 646](https://www.python.org/dev/peps/pep-0646/#use-case-1-specifying-shape-values).

The `TypeError` here is raised because `args` is not iterable in these cases. Singletons are not an issue in other subscriptable types because they are special-cased in their handling earlier in the function.

I did the error trapping this way initially b/c EAFP model. I can think of two other ways to handle offhand at this point in the code, both of which are LBYL, but more precise:

1. Switch to a `try: iter(args); except ...` handler. Would make it more specifically catch the iterable/non-iterable difference.

2. Switch to an explicit `isinstance(args, Number)` check. This would be a very focused check, and would not cover cases with non-numeric singletons. I believe neither `nptyping` nor PEP 646 uses non-numeric singletons, but I'm not 100% sure; and, who knows what's out there in the ecosystem, or will be.

I think I'm inclined to revise to (1).

Reply:
Sounds good for 1 👍